dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/sql.py

```diff
@@ -69,15 +70,25 @@ def __init__(
     ):
         super().__init__(name, TaskType.SQL, *args, **kwargs)
         self.sql = sql
+        self.param_sql_type = sql_type
         self.datasource_name = datasource_name
         self.pre_statements = pre_statements or []
         self.post_statements = post_statements or []
         self.display_rows = display_rows
 
     @property
     def sql_type(self) -> int:
+        """If sql type is specified by parameter"""
+        if (
+            self.param_sql_type == SqlType.SELECT
+            or self.param_sql_type == SqlType.NOT_SELECT
+        ):
+            return self.param_sql_type
```

Comment:
I think we should add some log before we retrue to user, such as https://github.com/apache/dolphinscheduler/blob/1f48601c759df8dae82e966bed1a346416c7449a/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/python.py#L81 but use `info` level